### DISCUSSION TITLE
Alerting "Active hours"

### DISCUSSION THREAD

**Discussion by YannickG1** [jertel/elastalert2#693] [Q&A] [answered]
Hi,

For a specific alerting requirement, we would like to only be alerted during "working hours". It is to trace that events are effectively consumed, but events only get fired during working hours. I was wondering if there is a  good way to do this.

So basically "if no matching document found for the last hour , alert me, but only during working hours because  we know  that during the night it is  a normal situation". I know I can use  the flatline alert but it would only work if I put the timeframe to 8 hours or something to make sure we don't get false positives during the night. This means during daytime we will only see the issue after 8 hours which is too long.

Any ideas/input is greatly appreciated!

Thanks!

**Comment by nsano-rururu** [comment, marked as answer]
https://elastalert2.readthedocs.io/en/latest/ruletypes.html#limit-execution
>tting aggregate_by_match_time.

>
>limit_execution[¶](https://elastalert2.readthedocs.io/en/latest/ruletypes.html#limit-execution)
>limit_execution: This option allows you to activate the rule during a limited period of time. This uses the cron format.

>
>For example, if you wish to activate the rule from monday to friday, between 10am to 6pm:

`limit_execution: "* 10-18 * * 1-5"`

**Reply by YannickG1** [reply]
Damn how could I have missed that. Sorry, I had consulted the docs but didn't find it when I looked. Thanks a lot !

**Reply by nsano-rururu** [reply]
Elastalert_hour_range
https://github.com/0xSeb/elastalert_hour_range
>Elastalert enhancement to make it possible to drop an alert if it's (or not) in a specific time range.
